### PR TITLE
EVG-7811: Configure Patch Page

### PR DESCRIPTION
**Pull request by tgrander**
https://jira.mongodb.org/browse/EVG-7810
Screen Shot 2020-04-22 at 8 46 28 AM

**Feature spec:** 
- Going to patch page for unactivated patch redirects to configure page
- Patch data is used for initial values on page
- Selecting build variant displays tasks of that variant
- Clicking on tasks checks them and updates task counts
- Select/Deselect all buttons to de/select all tasks

### Testing in staging
To test this feature in staging:
1. Submit a patch to staging server
2. Go to the patch page in new ui for submitted patch
You will be redirected to the configure page

### TO DO
Clicking schedule button schedules patch and redirects user to patch page
- https://jira.mongodb.org/browse/EVG-7811

### CODE REVIEW

File: cypress/integration/patch/configure-patch.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+          .then(() => {
+            tasks.forEach((task) => {
+              cy.get(`[data-cy=configurePatch-${task}]`).should("be.checked");
+            });
+          });
+      });
+    });
+  });
+  describe("Configuring a patch", () => {
+    it("Can update patch description by typing into `Patch Name` input field", () => {
```
Comment (SupaJoon) [inline cypress/integration/patch/configure-patch.ts:84]: This tests that we can type into the description input but it doesn't test that the description of the patch is actually updated. I think that will possible once the schedule button is hooked up.

Comment (tgrander) [reply cypress/integration/patch/configure-patch.ts:84]: i can add the test now

File: src/pages/configurePatch/ConfigurePatchCore.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    VariantTasksState
+  >(convertPatchVariantTasksToStateShape(variantsTasks));
+  const [descriptionValue, setdescriptionValue] = useState<string>(
+    patch.description
+  );
+  const onChangePatchName = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setdescriptionValue(e.target.value);
+  return (
+    <>
+      <StyledBody weight="medium">Patch Name</StyledBody>
```
Comment (john-m-liu) [inline src/pages/configurePatch/ConfigurePatchCore.tsx:46]: What's this label for? If it's referring to the description, I think it's not needed with the ghost text that's already there

Comment (tgrander) [reply src/pages/configurePatch/ConfigurePatchCore.tsx:46]: i proposed it in spruce channel to add clarity to the input. brian, ben, myself and sam all agreed. without it the input gets lost.

File: src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+  return (
+    <TabContentWrapper>
+      <Actions>
+        <Button variant={Variant.Primary}>Schedule</Button>
+        <ButtonLink
+          data-cy="configurePatch-selectAll"
+          onClick={onClickSelectAll}
+        >
+          Select All
```
Comment (SupaJoon) [inline src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx:93]: Ben recommended that these should be checkboxes with an indeterminate state

Comment (tgrander) [reply src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx:90]: the original design was incorrect and so john and i decided on another design. ben came up with the partial checkbox after the feature had already been implemented. i told him i can make a ticket to do the partial checkbox eventually.

Comment (SupaJoon) [reply src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx:90]: cool I made the ticket here: https://jira.mongodb.org/browse/EVG-7871

Comment (tgrander) [reply src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx:90]: bless

File: src/pages/configurePatch/ConfigurePatchCore.tsx (hunk 2 above)
Comment (john-m-liu) [reply src/pages/configurePatch/ConfigurePatchCore.tsx:46]: That works, but can we get rid of the ghost text then?

Comment (tgrander) [reply src/pages/configurePatch/ConfigurePatchCore.tsx:46]: done

**Pull request merged**